src/averageAges.js

```diff
@@ -15,11 +15,28 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const validAgeRange = [];
+  const filterFunction = (personGender, personDeathDate) => {
+    const genderFilter = (gender) => gender === 'm';
+    const centuryFilter = (death) => Math.ceil(death / 100) === century;
+
+    return century
+      ? genderFilter(personGender) && centuryFilter(personDeathDate)
+      : genderFilter(personGender);
+  };
+  const validPeople = people.filter((person) => {
+    return filterFunction(person.sex, person.died);
+  });
+
+  validPeople.forEach((person) => {
+    validAgeRange.push(person.died - person.born);
+  });
+
+  const averageAge = validAgeRange.reduce((acc, curr) => {
+    return acc + curr;
+  }, 0) / validAgeRange.length;
```

Comment:
This here can be a separate function, because you do the same when calculating women average age and age difference